exchanges/kraken/kraken_websocket.go

```diff
@@ -540,17 +542,30 @@ func (k *Kraken) wsProcessTrades(response []any, pair currency.Pair) error {
 		if !ok {
 			return errors.New("unidentified trade data received")
 		}
-		timeData, err := strconv.ParseFloat(t[2].(string), 64)
+		if len(t) < 4 {
+			return fmt.Errorf("unexpected trade data length: %s", t)
```

Comment:
This will need test coverage.
And implicit in that is having a defined err var.
Suggest that we've copy-pasta 🍝  `errParsingWSField` enough already and you should move it to `common.ErrParsingWSField`.
```suggestion
		if len(t) < 4 {
			return fmt.Errorf("%w; unexpected trade data length: %d", common.ErrErrParsingWSField, len(t))
```
Your test can then assert ErrorIs and ErrorContains 3 when you send a small slice through.

Reply:
Error added to common and all instances updated